torch/distributed/pipelining/_backward.py

```diff
@@ -248,6 +254,9 @@ def stage_backward_weight(
             grad_outputs=sum(param_group["grads"], tuple()),
             retain_graph=retain_graph,
         )
+        # release grad memory early after use
+        del param_group["grads"]
```

Comment:
can you say more about this? Does this meaningfully affect memory usage? What happened if we did not do this delete?

Reply:
ok, 'saves a few MB' means it is actually meaningful, even if its small.